src/components/data-management/ProjectDetails.jsx

```diff
@@ -357,37 +382,15 @@ const ProjectDetails = ({ width, height }) => {
             value={name}
           />
           <MetadataEditor
-            onReplaceEmpty={(value) => {
-              activeProject.samples.forEach(
-                (sampleUuid) => {
-                  if (
-                    !samples[sampleUuid].metadata[key]
-                    || samples[sampleUuid].metadata[key] === defaultNA
-                  ) {
-                    dispatch(updateSample(sampleUuid, { metadata: { [key]: value } }));
-                  }
-                },
-              );
-            }}
-            onReplaceAll={(value) => {
-              activeProject.samples.forEach(
-                (sampleUuid) => dispatch(updateSample(sampleUuid, { metadata: { [key]: value } })),
-              );
-            }}
-            onClearAll={() => {
-              activeProject.samples.forEach(
-                (sampleUuid) => dispatch(
-                  updateSample(sampleUuid, { metadata: { [key]: defaultNA } }),
-                ),
-              );
-            }}
+            onReplaceEmpty={(value) => replaceEmptyCells(value, key)}
+            onReplaceAll={(value) => replaceAllCells(value, key)}
+            onClearAll={() => clearAllCells(defaultNA, key)}
```

Comment:
is there an easy way for us instead of calling `updateSpeciesOrMetadata` from within each of those functions, to call it once and then pass the result to them directly?

Reply:
or, is there an easy, non leaky-abstraction way for us to use the same function for `onReplaceEmpty `, `onReplaceAll ` and for `onClearAll ` because they look almost the same?

Reply:
Refactored into `massUpdateCells`